File: PurchaseSale/Api/PurchaseSale.Entity/SaleDetail.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using OpenDataAccess.Entity;

namespace PurchaseSale.Entity
{
    [TableProperty(Name = "t_SaleDetail", PrimaryKey = "Id")]
    [RequestMethod(IsDelete = false, IsPut = false,IsPost=false, IsGet=false)]
    public class SaleDetail : EntityModel, IEntity
    {
        /// <summary>
        /// 主键
        /// </summary>
        public Guid Id { get; set; }
        /// <summary>
        /// 销售Id
        /// </summary>
        public Guid SaleId { get; set; }
        /// <summary>
        /// 商品Id
        /// </summary>
        public Guid ProductId { get; set; }
        /// <summary>
        /// 价格
        /// </summary>
        public decimal SillingPrice { get; set; }
        /// <summary>
        /// 进价
        /// </summary>
        public decimal BidPrice { get; set; }
        /// <summary>
        /// 折扣
        /// </summary>
        public decimal Discount { get; set; }
        /// <summary>
        /// 数量
        /// </summary>
        public float Number { get; set; }
        /// <summary>
        /// 金额
        /// </summary>
        public decimal Amount { get; set; }
    }

    [TableProperty(Name = "v_SaleDetail", PrimaryKey = "Id")]
    [RequestMethod(IsDelete = false, IsPost = false, IsPut = false)]
    public class ViewSaleDetail : SaleDetail
    {
        public string ProductTypeName { get; set; }
        public string ProductBrandName { get; set; }

        /// <summary>
        /// 名称
        /// </summary>
        public string ProductName { get; set; }
        /// <summary>
        /// 商品编号
        /// </summary>
        public string ProductCode { get; set; }
        /// <summary>
        /// 商品条形码
        /// </summary>
        public string ProductBarCode { get; set; }
        /// <summary>
        /// 型号
        /// </summary>
        public string Model { get; set; }
        /// <summary>
        /// 规格
        /// </summary>
        public string Spec { get; set; }
        /// <summary>
        /// 计量单位
        /// </summary>
        public string Unit { get; set; }
        /// <summary>
        /// 商品类型
        /// </summary>
        public Guid ProductTypeId { get; set; }
        /// <summary>
        /// 商品品牌
        /// </summary>
        public Guid ProductBrandId { get; set; }

        public byte SaleType { get; set; }
        public byte SaleStatus { get; set; }
        public string SaleTypeName { get; set; }
        public string SaleStatusName { get; set; }

        public string SaleCode { get; set; }
        public decimal Amount2 { get; set; }
        public decimal Discount2 { get; set; }
        public decimal BidAmount2 { get; set; }
        public decimal Profit { get; set; }
        public float ProfitRate { get; set; }

        public DateTime SaleDate { get; set; }
    }
}

File: WebHook/configs/entities/SaleDetail.js
module.exports = {
    Name: "SaleDetail",
    PrimaryKey: "Id",
    Properties: GetProperties(),
    SaleStatusDataSource: GetSaleStatusDataSource(),
    ProductTypeDataSource: GetProductTypeDataSource(),
    ProductBrandDataSource: GetProductBrandDataSource(),
}

function GetProperties() {
    return [
        GetProperty("Id", "Id"),
        GetProperty("SaleId", "销售Id"),
        GetProperty("SaleCode", "销售单号"),
        GetProperty("ProductId", "商品Id"),
        GetProperty("ProductName", "商品"),
        GetProperty("SillingPrice", "价格"),
        GetProperty("BidPrice", "进价"),
        GetProperty("Discount2", "折扣"),
        GetProperty("ProductTypeName", "类型"),
        GetProperty("ProductBrandName", "品牌"),
        GetProperty("Number", "数量"),
        GetProperty("Amount2", "金额"),
        GetProperty("Amount", "金额"),
        GetProperty("BidAmount2", "成本"),
        GetProperty("Profit", "利润"),
        GetProperty("SaleDate","销售日期"),
        GetProperty("ProfitRate", "利润率(%)"),
        GetProperty("SaleStatusName","状态")
    ]
}

function GetProperty(Name, Label) { return { Name, Label } }

function GetSaleStatusDataSource() {
    return [{ Value: 0, Text: "待提交" }, { Value: 1, Text: "已提交" }, { Value: 2, Text: "已存档" }, { Value: 3, Text: "已作废" }]
}

function GetProductTypeDataSource() {
    return {
        ValueName: "Id",
        TextName: "Name",
        StateName: "ProductTypes",
        ServiceName: "SaleService",
        ActionName: "GetProductTypes",
        IsRefresh: true,
        Payload: {}
    }
}

function GetProductBrandDataSource() {
    return {
        ValueName: "Id",
        TextName: "Name",
        StateName: "ProductBrands",
        ServiceName: "SaleService",
        ActionName: "GetProductBrands",
        IsRefresh: true,
        Payload: {}
    }
}
